From fd8f052e034476b2bd2260a0cfb84e547faeb818 Mon Sep 17 00:00:00 2001
From: Bingwu Zhang <xtex@aosc.io>
Date: Sat, 8 Feb 2025 12:05:04 +0800
Subject: [PATCH 4/4] [loong64] Honor existing LASX/LSX settings
X-Developer-Signature: v=1; a=openpgp-sha256; l=869; i=xtex@aosc.io;
 h=from:subject; bh=Kx9QEBi2/PSAbavbNIZyRebIqDVHB+qj/P4JUxAoHRo=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDOnLZbKLZnjZFkUcFlApzvmW5viXPzSkv0fjxMozW8QbU
 za+P7Wgo5SFQYyLQVZMkaXIsMGbVSedX3RZuSzMHFYmkCEMXJwCMJEPcxkZVn07eNZ/yuKW0ymM
 i3KvdtqHTbQ87Scxs5f7wqO77pPufmBkuNL0yvjc36/i3Q3M3Pt41a6UPeOZY8oSt6xH9U5Z75/
 dfAA=
X-Developer-Key: i=xtex@aosc.io; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

Change-Id: I1e6183b987002ed09621b111a6012a0f186233f1
Signed-off-by: Bingwu Zhang <xtex@aosc.io>
---
 src/opts/SkRasterPipeline_opts.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/opts/SkRasterPipeline_opts.h b/src/opts/SkRasterPipeline_opts.h
index a0b0a3de8bd9..84650c34377a 100644
--- a/src/opts/SkRasterPipeline_opts.h
+++ b/src/opts/SkRasterPipeline_opts.h
@@ -69,7 +69,7 @@ using NoCtx = const void*;
 
 #if defined(SKRP_CPU_SCALAR) || defined(SKRP_CPU_NEON) || defined(SKRP_CPU_HSW) || \
         defined(SKRP_CPU_SKX) || defined(SKRP_CPU_AVX) || defined(SKRP_CPU_SSE41) || \
-        defined(SKRP_CPU_SSE2)
+        defined(SKRP_CPU_SSE2) || defined(SKRP_CPU_LASX) || defined(SKRP_CPU_LSX)
     // Honor the existing setting
 #elif !defined(__clang__) && !defined(__GNUC__)
     #define SKRP_CPU_SCALAR
-- 
2.48.1

